### ISSUE TITLE
Nav Unification: Error Clicking "WooCommerce" 

### ISSUE THREAD

**Issue by rosepajaroja** [Automattic/wp-calypso#50059]
### Steps to reproduce the behavior

1. Click "WooCommerce".
2. See error "Cannot load woocommerce."

#### What I expected to happen

Load the WooCommerce dashboard

#### What actually happened
Error prompt: "Cannot load woocommerce."

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.

Tested on two test sites (Business and eCommerce plans)

https://woowebinartest.wpcomstaging.com/
https://atomicv2testsite4rosep022019.wpcomstaging.com

https://www.loom.com/share/2ae8fc9a4a36414e96d627757a75700f

**Comment by mmtr** [comment]
I think this has been already fixed by https://github.com/woocommerce/woocommerce/pull/29088 and https://github.com/woocommerce/woocommerce-admin/pull/6310, however they won't be released until WooCommerce 5.1.

In https://github.com/Automattic/wp-calypso/issues/50065 @mattsherman is investigating how can we have these fixes in place before the next WooCommerce release.

**Comment by mattsherman** [comment]
Hmm... this one was **supposed** to be fixed by https://github.com/woocommerce/woocommerce/pull/29088 and https://github.com/woocommerce/woocommerce-admin/pull/6310. 

However, I just tested this again on a few sites, and this is *not* fixed. I am not sure how we missed this one when testing before those were merged. I could swear that was fixed (I even mentioned it in 6310) -- I know, famous last words.

/cc @psealock @getdave

**Comment by psealock** [comment]
>Hmm... this one was supposed to be fixed by woocommerce/woocommerce#29088 and woocommerce/woocommerce-admin#6310.

😓

I confirmed this is caused by the conditional added in https://github.com/woocommerce/woocommerce-admin/pull/6310/ that applies the new logic for `version_compare( WC_VERSION, '5.1', '>=' )`. The condition must evaluate to `false` when the version is `5.1.0-beta.1` because I hard coded to true and re-uploaded `5.1.0-beta.1` and things are working as they should.

I wouldn't mind another set of eyes on this because when I log this out it DOES evaluate to `true`

```
error_log( version_compare( '5.1.0-beta.1', '5.1', '>=' ) );
```

Perhaps there is something else going on because WooCommerce Core uses `Automattic\Jetpack\Constants::get_constant( 'WC_VERSION' )` to access that global.

**Comment by mattsherman** [comment]
> I confirmed this is caused by the conditional added in [woocommerce/woocommerce-admin#6310](https://github.com/woocommerce/woocommerce-admin/pull/6310) that applies the new logic for `version_compare( WC_VERSION, '5.1', '>=' )`. The condition must evaluate to `false` when the version is `5.1.0-beta.1` because I hard coded to true and re-uploaded `5.1.0-beta.1` and things are working as they should.

@psealock Do you already have a new PR to for us to test and get merged?

**Comment by psealock** [comment]
>@psealock Do you already have a new PR to for us to test and get merged?

I took a deeper look and was way off on the version thing. Its a question about how WP constructs urls for menu items v. how Jetpack creates the item's url.

>Hmm... this one was supposed to be fixed by woocommerce/woocommerce#29088 and woocommerce/woocommerce-admin#6310.

I think it was always broken. The first submenu item `Home` works as expected, but the parent points to `woocommerce` still.

In wc-admin, we manipulate the order of the `$submenu['woocommerce']` object to get the desired url for WooCommerce:

https://github.com/woocommerce/woocommerce-admin/blob/d08445d90cae5eefc654329953ff09ec750d564c/src/Features/Homescreen.php#L116-L117

In other words, WC Admin changes the array of submenu items such that `Home` is first. This seems to be the key logic that changes its **parent item's url** in a way thats not entirely clear to me. I do see a pattern in all wp-admin links where parent items with submenus share the same url with the first of its submenu items. Is this a hack that we used to get the `WooCommerce` menu item to point to wc-admin's Home screen? @joshuatf by chance, would you be able to shed some light on this?

In Calypso, [Jetpack's admin-menu API endpoint is used to gather menu items and construct urls](https://github.com/Automattic/jetpack/blob/5da9fba166a08ea826de133f411ab3c3ce68d06a/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php#L311-L356). What constructs that url is line 349 in that file:

```php
$url = add_query_arg( array( 'page' => $url ), admin_url( 'admin.php' ) );
```

Doing some digging around, a menu item taking on its first child's url appears [to be "how things work"](https://wordpress.org/support/topic/top-menu-item-with-set-url-redirecting-to-first-submenu-url-instead/#post-9938652). If an item in the `$menu` global always shares its url with its first `$submenu` item, then I think the fix lies in Jetpack which should replicate that logic. 

cc @mmtr since you authored [the last change to the API](https://github.com/Automattic/jetpack/pull/18898) (thank you!).

**Comment by mmtr** [comment]
> Doing some digging around, a menu item taking on its first child's url appears to be "how things work". If an item in the $menu global always shares its url with its first $submenu item, then I think the fix lies in Jetpack which should replicate that logic

That sounds right, I confirmed it by looking at the code in Core that renders the menu and checked that the URL of the first submenu is always used in the parent: https://core.trac.wordpress.org/browser/trunk/src/wp-admin/menu-header.php?rev=49193#L152

Will try to prepare a fix for the `admin-menu` endpoint to replicate that behavior.

**Comment by mattsherman** [comment]
@psealock @mmtr Thanks for digging into this and arriving at a solution. Excellent! I have tested and verified this fix works well with WooCommerce 5.1 Beta 1 and the JP 9.5 release candidate on one of my test sites!
